drivers/gpio/gpio_andes_atcgpio100.c

```diff
@@ -24,20 +24,13 @@
 #define DT_DRV_COMPAT andestech_atcgpio100
 
 /* Andes ATCGPIO100 register definition */
-#define REG_IDR  0x00 /* ID and Revision reg.           */
```

Comment:
Why are definitions being removed from this file?

I would suggest leaving them

Reply:
Because we were asked to remove unneeded definitions when adding our spi driver upstream. 
So isn't it a general rule for drivers?

Reply:
Dead code is one thing - register descriptions are another. Having `#define`s for otherwise undocumented register areas is probably better than nothing. What do you think @henrikbrixandersen ?

Reply:
I'd leave them in for completeness.